### PR TITLE
[KO] Update translation

### PR DESCRIPTION
**Pull request by marocchino**
@Dalgona, @malkoG, @devleoper, @riseshia review please.

FYI. I use this command for diff:

```shell
git diff f1f3b3d..f8baebe --word-diff -- lessons/
```

### CODE REVIEW

**Comment by marocchino** [comment]
relate #711

File: ko/lessons/basics/enum.md [hunk 1]
```diff
@@ -12,9 +12,30 @@ lang: ko
 
 ## Enum
 
-`Enum` 모듈은 우리가 지난 강의에서 배웠던 컬렉션들을 이용하기 위한 100가지 이상의 함수들을 포함하고 있습니다.
+`Enum` 모듈은 우리가 지난 강의에서 배웠던 컬렉션들을 이용하기 위한 100개 조금 넘는 함수들을 포함하고 있습니다.
 
-이 강의에서는 그 함수들 중에서도 일부분을 다룰 것입니다. 전체적인 부분을 학습하실 분들께서는 공식 [`Enum`](http://elixir-lang.org/docs/stable/elixir/Enum.html) 문서를 보시면 되겠습니다; 지연 열거(lazy enumeration)를 이용하시려면 [`Stream`](http://elixir-lang.org/docs/stable/elixir/Stream.html) 모듈을 이용해보세요.
+이 강의에서는 그 함수들 중에서도 일부분을 다룰 것입니다. 하지만 실제로 시험해 볼 수 있습니다.
```
Comment (riseshia) [inline ko/lessons/basics/enum.md]: Do we need `하지만` here?

File: ko/lessons/basics/enum.md [hunk 2]
```diff
@@ -12,9 +12,30 @@ lang: ko
...
+IEx에서 조금 시험에 보죠.
```
Comment (riseshia) [inline ko/lessons/basics/enum.md]: `시험에` -> `시험해`

File: ko/lessons/basics/enum.md [hunk 3]
```diff
@@ -12,9 +12,30 @@ lang: ko
...
+all?/1
+all?/2
+any?/1
+any?/2
+at/2
+at/3
+...
+```
+
+이 것을 사용해 아주 많은 기능이 있고 기능마다 분명한 이유가 있음이 자명해 졌습니다.
```
Comment (riseshia) [inline ko/lessons/basics/enum.md]: (maybe) `이 것을 사용해` -> `이를 통해`

File: ko/lessons/basics/enum.md [hunk 4]
```diff
@@ -12,9 +12,30 @@ lang: ko
...
+열거형은 함수형 프로그래밍의 핵심이며, 놀랍도록 유용합니다.
+By leveraging it combined with other perks of Elixir, such as documentation being a first class citizen as we just saw, it can be incredibly empowering to the developer as well.
```
Comment (riseshia) [inline ko/lessons/basics/enum.md]: untranslated XD

File: ko/lessons/basics/modules.md [hunk 5]
```diff
@@ -252,3 +252,5 @@ end
 Ecto.Migration를 어떤 모듈에서 사용할 것이라고 선언(use)하면 `Ecto.Migration.__using__/1` 매크로가 import를 실행시켜 사용(use)하는 모듈에 Ecto.Migration 모듈 안에 있는 함수와 매크로를 불러오도록(import) 합니다. Ecto가 어떻게 작동할 지 정할 수 있는 것처럼 보이는 모듈 속성을 정의하기도 합니다.
 
 다시 말하자면, use 매크로는 해당 모듈의 `__using__/1` 매크로를 실행하기만 합니다. `__using__/1` 매크로가 정확히 무엇을 하는지 이해하려면 해당하는 부분의 코드를 읽어야 합니다.
+
+**주의**: `quote`, `alias`, `use`, `require`는 [메타프로그래밍](../advanced/metaprogramming)에서 사용한 메크로입니다.
```
Comment (riseshia) [inline ko/lessons/basics/modules.md]: `메타프로그래밍`? or  `메타 프로그래밍`?

Comment (riseshia) [inline ko/lessons/basics/modules.md]: `메크로` -> `매크로`

File: ko/lessons/specifics/mnesia.md [hunk 6]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.write({Person, 5, "Hans Moleman", "Ex-Mayor"})
+...>   end
+...> )
+```
+
+## 인덱스 사용하기
+
+Mnesia 키가 아닌 컬럼에 인덱스를 지원하고 이 데이터에 대해 쿼리를 던질 수 있습니다. `Person` 테이블에 `:job` 컬럼에 인덱스를 추가해 봅시다.
```
Comment (riseshia) [inline ko/lessons/specifics/mnesia.md]: `테이블에` -> `테이블의`

File: ko/lessons/specifics/mnesia.md [hunk 7]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+
+```elixir
+iex> Mnesia.add_table_index(Person, :job)
+{:atomic, :ok}
+```
+
+`Mnesia.create_table/2`의 결과와 비슷해 보입니다.
+
+ - `{:atomic, :ok}` 함수가 성공적으로 실행 됨
```
Comment (riseshia) [inline ko/lessons/specifics/mnesia.md]: (maybe) `실행 됨` -> `실행됨`

File: ko/lessons/specifics/mnesia.md [hunk 8]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+ - `{:aborted, Reason}` 함수가 실패함
+
+일반적으로 키가 이미 존재하면, `{:already_exists, table, attribute_index}` 형식의 응답이 돌아옵니다. 두 번째 이 인덱스를 넣으려 했다면, 이런 결과를 받게 됩니다.
```
Comment (riseshia) [inline ko/lessons/specifics/mnesia.md]: (maybe) `두 번째 이 인덱스를 넣으려 했다면` -> `이 인덱스를 한번 더 넣으려 했다면`

File: ko/lessons/specifics/mnesia.md [hunk 9]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.match_object({Person, :_, "Marge Simpson", :_})
+...>   end
+...> )
+{:atomic, [{Person, 4, "Marge Simpson", "home maker"}]}
+```
+
+`Mnesia.select/2` 함수는 Elixir(나 Erlang)의 어떤 연산자나 함수로도 사용자 정의 쿼리를 정의할 수 있습니다. 키가 3보다 큰 모든 레코드를 찾는 예제를 살펴봅시다.
```
Comment (riseshia) [inline ko/lessons/specifics/mnesia.md]: (maybe) `함수로도` -> `함수로든`

File: ko/lessons/specifics/mnesia.md [hunk 10]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+
+풀어 봅시다. 첫 번째 인자는 테이블 `Person` 입니다. 두 번째 인자는 `{match, [guard], [result]}` 형식입니다.
+
+- `match`는 `Mnesia.match_object/1` 함수에 넘기는 것과 동일합니다. 하지만 쿼리의 나머지에 사용하는 파라미터의 위치를 지정하는 특수 애텀 `:"$n"`에 주의하세요.
+- `guard` 리스트는 적용할 가드 함수를 지정하는 튜플의 리스트입니다. 이 경우에는 `:>` (크거나 같음) 빌트인 함수, 위치 파라미터 `:"$1"`, 상수 `3`이 인자로 사용되었습니다.
+- `result` 리스트는 쿼리가 반환하는 필드 리스트입니다. 위치 파라미터 형식에서 특수 에텀 `:"$$"`은 참조할 수 있는 모든 필드를 나타냅니다. `[:"$1", :"$2"]`로 처음 두 필드만 반환하거나 `[:"$$"]`로 모든 필드를 반환할 수 있습니다.
+
+더 자세한 내용은 [Erlang Mnesia 문서의 select/2](http://erlang.org/doc/man/mnesia.html#select-2)를 참조하세요.
+
+## 데이터 초기화와 이관
```
Comment (devleoper) [inline ko/lessons/specifics/mnesia.md]: Question: Is the word _migrate/migration_ frequently translated to **이관** in Korean? That feels like moving all data from a database into another (_transfer_). In this context, is this _migration_ used as those of Rails? then choosing the popular word for _migrate_ in those community.

File: ko/lessons/specifics/mnesia.md [hunk 11]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+
+모든 소프트웨어 제품이 그렇듯이 소프트웨어를 업그레이드하고 데이터베이스에 저장된 데이터를 이관해야할 때가 있습니다. 예를 들어 앱을 v2로 버전업하면서 `Person` 테이블에 `:age` 컬럼을 추가해야 한다고 해봅시다. 테이블이 이미 있으므로 `Person` 테이블을 만들수는 없지만 변환할 수는 있습니다. 이를 위해서는 테이블을 생성할 때 무엇을 할 수 있는 지, 언제 변환해야 하는 지 알아야 합니다. `Mnesia.table_info/2` 함수를 사용해 현재 테이블의 구조를 검색하고 `Mnesia.transform_table/3`로 새 구조로 변환할 수 있습니다.
+
+아래에 있는 코드는 다음 로직을 구현합니다.
+* v2 인자 `[:id, :name, :job, :age]`로 테이블 생성
+* 생성 결과를 처리
+  * `{:atomic, :ok}`: `:job`, `:age`에 인덱스를 생성해 테이블 초기화
+  * `{:aborted, {:already_exists, Person}}`: 현재 테이블에 어떤 인자가 있고 올바르게 동작하는 지 확인
```
Comment (devleoper) [inline ko/lessons/specifics/mnesia.md]: > 올바르게 **동작하는지**

Korean spacing sometimes bothers.

File: ko/lessons/specifics/mnesia.md [hunk 12]
```diff
@@ -174,3 +181,118 @@ iex> data_to_read = fn ->
...
+    * v1의 목록(`[:id, :name, :job]`)이라면 모든 레코드의 나이를 21를 설정하고 `:age`에 새 인덱스를 설정해 변환
+    * v2의 목록이라면 아무것도 하지 않음
+    * 다른 결과면 종료
+
+`Mnesia.transform_table/3` 함수는 인자로 테이블의 이름, 레코드를 옛날 것에서 새것으로 변환하는 함수, 새 인자의 리스트를 받습니다.
```
Comment (devleoper) [inline ko/lessons/specifics/mnesia.md]: > `Mnesia.transform_table/3` 함수는 테이블 이름, 이전 레코드를 새 레코드로 변환하는 함수, 새 인자의 리스트를 인자로 받습니다.

Suggestion for better readability:
 - Moving **인자로** backward
 - Omitting **것**.

File: ko/lessons/advanced/typespec.md [hunk 13]
```diff
@@ -34,9 +34,10 @@ end
 ```
 
 전부 다 괜찮아 보이고, 함수를 호출하면 올바른 값이 리턴되겠지요. 하지만 `Enum.sum` 함수는 `@spec`에서 예상했던 `integer`가 아니라 `number`를 리턴합니다. 이런 부분에서 버그가 생겨날 수 있어요! 코드를 정적 분석해주는 Dialyzer 같은 도구를 사용해서 이런 종류의 버그를 찾아낼 수 있습니다. 이런 정적 분석 도구를 사용하는 법에 대해서는 다른 수업에서 다루어 보겠습니다.
- 
+
 ## 커스텀 타입
-스펙을 작성하는 것도 좋지만, 때로는 우리들이 구현한 함수가 간단한 함수나 컬렉션보다 복잡한 자료 구조를 처리해야 할 수도 있습니다. 이런 함수를 `@spec`으로 정의한다면, 다른 개발자들이 이해하거나 수정하기가 굉장히 힘들어질 수 있습니다. 종종 함수가 많은 파라메터를 필요로 하거나, 복잡한 데이터를 리턴해야 할 때가 있습니다. 하지만 파라메터 목록이 길어질수록 코드의 품질이 떨어질 가능성도 점점 커집니다. Ruby나 Java 같은 객체 지향 언어를 사용했더라면, 간편하게 클래스를 구현해서 문제를 해결할 수 있었을 것입니다. Elixir에서는 클래스가 없고, 대신에 타입을 정의해서 언어를 확장할 수 있습니다.
```
Comment (marocchino) [inline ko/lessons/advanced/typespec.md]: Yes, 'cause Original text removes very.

File: ko/lessons/specifics/mnesia.md (hunk 9 above)
Comment (marocchino) [inline ko/lessons/specifics/mnesia.md]: Well, custom means, it is not prepared one.
We can replace that with 직접.
Is that OK for you?

> `Mnesia.select/2` 함수로 Elixir(나 Erlang)의 연산자, 함수를 사용해 직접 쿼리를 정의할 수 있습니다.

File: ko/lessons/specifics/mnesia.md (hunk 10 above)
Comment (marocchino) [inline ko/lessons/specifics/mnesia.md]: I use this terms: 
- migrate: 이관
- transfer: 전송
- transform: 변환
I know 이관 came form Japanese word 移管.
But, I do not know frequently. 

And for 2nd question, Yes.

**Pull request merged**